be/src/storage/lake/txn_log_applier.cpp

```diff
@@ -391,10 +391,10 @@ class PrimaryKeyTxnLogApplier : public TxnLogApplier {
                              << ", base version: " << txn_meta.visible_version() << ", new version: " << _new_version;
                 return Status::Corruption("mismatched snapshot version and new version");
             }
-        } else if (txn_meta.snapshot_version() - txn_meta.data_version() + txn_meta.visible_version() != _new_version) {
+        } else if (txn_meta.snapshot_version() - txn_meta.data_version() + _base_version != _new_version) {
```

Comment:
`_base_version` is the same in a transactionStateBatch, either disable batch publish of replication transactionState or make a compatibility judgement so that batch publish still works.